### ISSUE TITLE
Unresolved dependency error when using 2.5.1 version

### ISSUE THREAD

**Issue by mckrpk** [algolia/instantsearch-android#228]
It helps to upgrade to 2.8.0 but I would rather not since it requires minSdk 21 while 2.5.1 requires 17.

It's worth to note that issue started without any change from my side since around Feb 11 2021.

Gradle 6.1.1
Android Gradle Plugin: 4.0.2
Kotlin 1.3.72

```
* What went wrong:
Could not determine the dependencies of task ':app:compileStagingReleaseJavaWithJavac'.
> Could not resolve all task dependencies for configuration ':app:stagingReleaseCompileClasspath'.
   > Could not resolve com.algolia:algoliasearch-client-kotlin-common:1.4.0.
     Required by:
         project :app > com.algolia:instantsearch-android:2.5.1
      > Unable to find a matching variant of com.algolia:algoliasearch-client-kotlin-common:1.4.0:
          - Variant 'metadata-api' capability com.algolia:algoliasearch-client-kotlin-common:1.4.0:
              - Incompatible attributes:
                  - Required org.gradle.usage 'java-api' and found incompatible value 'kotlin-api'.
                  - Required org.jetbrains.kotlin.platform.type 'androidJvm' and found incompatible value 'common'.
              - Other attributes:
                  - Required com.android.build.api.attributes.BuildTypeAttr 'release' but no value provided.
                  - Found org.gradle.status 'release' but wasn't required.
                  - Required version 'staging' but no value provided.
```

**Comment by Aallam** [comment]
Hi @mckrpk,

I highly recommend updating your min sdk to 21, this is related to TLS 1.2 support, which is the default since API 21, clients using the versions below API 21 may face some issues using Algolia.

That's being said, the error seems to be coming from the Gradle version update, rolling back your Gradle version might resolve your issue.

**Comment by mckrpk** [comment]
Hi @Aallam, 

Yeah OkHttp also bumped minSdk to 21 because of [this](https://developer.squareup.com/blog/okhttp-3-13-requires-android-5/).

It's strange but I didn't update Gradle, plugin or anything before the error started to happen. I noticed however that that `algoliasearch-client-kotlin-common` was published to [Maven](https://mvnrepository.com/artifact/com.algolia/algoliasearch-client-kotlin-common/1.4.0) on the date when it started failing. Not sure what it means but maybe that's not a coincidence.

Anyway I got around it by excluding transitive dependency causing error and adding it from local jar.
```
implementation("com.algolia:instantsearch-android:2.5.1") {
    exclude group: 'com.algolia', module: 'algoliasearch-client-kotlin-common'
}
implementation files("libs/algoliasearch-client-kotlin-common-1.4.0.jar")
```
and adding this to make debug variants work:
```
debug {
    matchingFallbacks = ['release']
}
```
This should do until we're ready to drop support for Android below 5.0 or decide to only use [algoliasearch-client-kotlin](https://github.com/algolia/algoliasearch-client-kotlin).
